css/catalog.scss

```diff
@@ -32,3 +32,7 @@
   margin-top: 3%;
   margin-bottom: 3%;
 }
+
+footer.footer {
```

Comment:
Well, rule specificity is computed through both selector specificity and order of inclusion if specificity is equivalent, so you always have to _account for_ it, if not _rely on_ it.

If your use case is overriding site-wide generic rules for a specific type of page, it seems (to me) pretty logical that order of inclusion should always be from most generic to most specific, and that you may rely on order of inclusion. At least, I don't find this much more fragile than relying on a non-semantic, redundant class name.

If you don't feel at ease with it, here are a few ideas.
If you want to increase selector specificity for generic elements, a clean way to do it is to add `class="{{ page.layout }}"` to the `<body>`, which will give you a very clear and readable `.$layout footer` selector.
Alternatively, if you really prefer to add a class to the `<footer>` do it, but don't use `footer` for the class name, that really doesn't mean anything and will only be clutter if you ever remove the dependent CSS rules.
Finally, if you want to increase rule specificity for a very specific case, you can use `! important`. This is a never-ending race though, so make sure you use it for “final” rules.